src/server/src/main/java/io/cassandrareaper/AppContext.java

```diff
@@ -49,6 +49,9 @@
   public JmxConnectionFactory jmxConnectionFactory;
   public ReaperApplicationConfiguration config;
   public MetricRegistry metricRegistry = new MetricRegistry();
+  public String localNodeAddress;
+  public String localClusterName;
+  public String localDatacenter;
```

Comment:
for better encapsulation:
if the three new private methods added to `Heart` are moved down to `MetricsService`
then these three fields can also be moved to `MetricsService` (and made private).

Reply:
I would disagree here.
We can move them as static vars in `ClusterFacade` if you think it belongs there more than in `AppContext`, but I feel that's not what you're going after.
Duplicating the variables in both `ClusterFacade` and `MetricsService` would only add possible confusion and we're going to have to add it in `SegmentRunner` too. That's 3 places to init/maintain the same variable, which adds tech debt IMHO.

Reply:
Let me take a closer look…

Reply:
`SegmentRunner` does not use any of these three fields. (Usage was from a method that can be removed: https://github.com/thelastpickle/cassandra-reaper/pull/622/files#r262444370

Reply:
So…
 - `localClusterName` and `localDatacenter` are only used in `MetricsService`, and so can be moved to, and initialised, there. 
 - `localNodeAddress` is used in many places, but only when in side-car mode.
 - define it as `volatile String localNodeAddress = null;`
 - add the following getter…
```
public String getLocalNodeAddress() {
    Preconditions.checkState(context.config.isInSidecarMode());
    return localNodeAddress;
}
```

Reply:
@adejanovski what's the status on this?

Reply:
it's done, pushed and tested